### PR TITLE
[Freebox] Various little changes

### PR DESCRIPTION
**Pull request by lolodomo**
Properties handling
Static code analysis
Exported FreeboxDataListener
Discovery service activate/deactivate

Replaces #2872

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@martinvw : can you review this PR that mainly improves the code ?

File: addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/handler/FreeboxHandler.java [hunk 1]
```diff
@@ -347,25 +344,16 @@ public void initialize() {
         }
 
         Map<String, String> properties = editProperties();
-        boolean update = false;
-        if (StringUtils.isNotEmpty(apiBaseUrl)
-                && !apiBaseUrl.equals(properties.get(FreeboxBindingConstants.API_BASE_URL))) {
-            update = true;
+        if (apiBaseUrl != null && !apiBaseUrl.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/handler/FreeboxHandler.java]: Kind of copy&paste from another binding code.
I will restore the use of StringUtils.isNotEmpty.

[the remark this answers is not in the document]
Comment (martinvw) [reply addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/handler/FreeboxHandler.java]: @lolodomo 

> Why stop using `StringUtils.isNotEmpty`?

**Comment by lolodomo** [comment]
@martinvw : updated

**Pull request merged**